src/Core/Services/MetadataProviders/DatabaseObjectConverter.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Config.DatabasePrimitives;
+
+namespace Azure.DataApiBuilder.Core.Services.MetadataProviders
+{
+    /// <summary>
+    /// This is a converter to serialize and deserialize the DatabaseObject
+    /// this adds a typename field at the object level, this is used when you deserialize the different child objects : DatabaseTable,
+    /// DatabaseView as DatabaseObject is an abstract class
+    /// This is also required as there is no TypeNameHandling support in System.text.Json and we need to explicitly add a TypeName field to distinguish
+    /// </summary>
+    public class DatabaseObjectConverter : JsonConverter<DatabaseObject>
+    {
+        public override DatabaseObject Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            using (JsonDocument document = JsonDocument.ParseValue(ref reader))
+            {
+                JsonElement root = document.RootElement;
+                string typeName = root.GetProperty("TypeName").GetString() ?? throw new JsonException("TypeName is missing");
+                Type concreteType = GetTypeFromName(typeName);
+
+                if (concreteType == null)
+                {
+                    throw new JsonException($"Unknown type: {typeName}");
+                }
+
+                DatabaseObject objA = (DatabaseObject)JsonSerializer.Deserialize(document, concreteType, options)!;
+
+                return objA;
+            }
+        }
+
+        public override void Write(Utf8JsonWriter writer, DatabaseObject? value, JsonSerializerOptions options)
+        {
+            writer.WriteStartObject();
+
+            // Add TypeName
+            writer.WriteString("TypeName", value!.GetType().AssemblyQualifiedName);
```

Comment:
Didnt understand this part where databaseObject is a nullable type but we explicitly say not null on line 43. Could we add a comment here explaining why it cant be null?

Reply:
Updated the comment in code

This line will : 
Add TypeName property in DatabaseObject object that we are serializing based on its type. The DatabaseObject can be of many types : DatabaseTable, DatabaseView etc. We add this property to differentiate between them in the dictionary. This extra property gets used in deserialization above. For example if object is DatabaseTable then we need to add "TypeName": "Azure.DataApiBuilder.Config.DatabasePrimitives.DatabaseTable, Azure.DataApiBuilder.Config, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null we cannot perform GetType() on a null object hence the check.